src/duckling/project/project.service.ts

```diff
@@ -252,11 +252,56 @@ export class ProjectService {
      * @param metaDataFile The name of the meta data file stored in the .duckling/ folder (excluding file type)
      * @return the path that can be used to load the meta data file
      */
-    getUserMetaDataPath(metaDataFile : string) : string {
+    getUserMetaDataPath(metaDataFile: string): string {
         return this._pathService.join(this.userMetaDataDir, metaDataFile + ".json");
     }
 
-    private async _parseMapJson(json : any, key : string) {
+    async runExistingCodeMigration(migrationName: string, options: any): Promise<void> {
+        await this._updateVersionFileWithExistingCodeMigration(migrationName, options);
+        await this._updateEntitySystemWithExistingCodeMigration(migrationName, options);
+        await this._updateProjectVersionInfo();
+    }
+
+    private async _updateVersionFileWithExistingCodeMigration(migrationName: string, options: any) {
+        let versionFile = await this.openVersionFile();
+        this._migrationService.updateVersionFileWithExistingCodeMigration(versionFile, migrationName, options);
+        this.saveVersionFile(versionFile);
```

Comment:
We shouldn't save the version file until the save button is clicked for the open map. This is how the collision-types file works, and this will work for this migration strategy.